src/main/scala/com/linkedin/feathr/offline/util/FeaturizedDatasetMetadata.scala

```diff
@@ -1,4 +1,10 @@
 package com.linkedin.feathr.offline.util
 
-case class FeaturizedDatasetMetadata(meta: Map[String, String] = Map()) {
+import com.linkedin.feathr.common.Header
+/**
+ * The metadata for FeaturizedDataset
+ * @param meta extra metadata
+ * @param header feature type header info
+ */
+case class FeaturizedDatasetMetadata(meta: Map[String, String] = Map(), header: Option[Header] = None ) {
```

Comment:
Should be more docs about what this means; what a featurized-dataset is? (Do we have docs about it in this public version of feathr? Should we just call it as feathr's output dataset that includes info about the tensor types of joined features...?)

Also, what does "extra metadata" mean; what kinds of things is the map allowed/expected to contain, what are the keys, values, etc.